test/_e2e/trigger_http_test.go

```diff
@@ -37,7 +38,7 @@ func (f FunctionHttpResponsivenessValidator) Validate(t *testing.T, project Func
 	if f.contentType != "" {
 		req.Header.Add("Content-Type", f.contentType)
 	}
-	client := &http.Client{}
+	client := &http.Client{Timeout: time.Second * 15}
```

Comment:
Interesting addition.  Does your debugging suggest that the reason we see tests flaking is (as we guessed) a timeout?

Reply:
Well, actually I was suspecting it was using the previous version (despite the new one report as ready). So changed the mechanistic to verify the new revision availability
https://github.com/knative-sandbox/kn-plugin-func/pull/970/files#diff-231b1bc48c9920dbdf67aae9fa6582460b37c59b89c1b226bc97a715137522d3R42

Here it was about to let it a explicit a reasonable time we expect a function to respond (even those I think that number is little high for a function).